File: src/main/java/org/padler/gradle/minify/MinifyPlugin.java
package org.padler.gradle.minify;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

public class MinifyPlugin implements Plugin<Project> {

    public static final String EXTENSION_NAME = "minification";
    public static final String TASK_NAME = "minify";

    @Override
    public void apply(Project project) {
        project.getExtensions().create(EXTENSION_NAME, MinifyPluginExtension.class);
        MinifyTask minify = project.getTasks().create(TASK_NAME, MinifyTask.class);
        minify.setGroup("build setup");
    }

}

File: src/main/kotlin/org/padler/gradle/minify/MinifyPlugin.kt
package org.padler.gradle.minify

import org.gradle.api.Plugin
import org.gradle.api.Project

open class MinifyPlugin : Plugin<Project> {

    override fun apply(project: Project) {
        val extension = project.extensions.create("minification", MinifyPluginExtension::class.java)
        project.afterEvaluate {
            if (extension.addDefaultCssMinifyTask) {
                val task = project.tasks.create("jsMinify", JsMinifyTask::class.java)
                extension.defaultJsMinifyTaskContext.applyOn(task)
            }
            if (extension.addDefaultCssMinifyTask) {
                val task = project.tasks.create("cssMinify", CssMinifyTask::class.java)
                extension.defaultCssMinifyTaskContext.applyOn(task)
            }
        }
    }
}
